Library/Homebrew/cli/args.rb

```diff
@@ -54,110 +49,47 @@ def freeze_processed_options!(processed_options)
       end
 
       def named
-        named_args || []
+        named_args || NamedArgs.new
       end
 
       def no_named?
         named.blank?
       end
 
       def formulae
-        require "formula"
-
-        @formulae ||= (downcased_unique_named - casks).map do |name|
-          Formulary.factory(name, spec, force_bottle: force_bottle?, flags: flags_only)
-        end.uniq(&:name).freeze
+        named.to_formulae
```

Comment:
For a follow-up PR: could add commented-out `odeprecated` to all these methods so I can enable them for 2.5.0?